src/main/java/org/openstreetmap/atlas/checks/validation/areas/OverlappingAOIPolygonCheck.java

```diff
@@ -0,0 +1,237 @@
+            "leisure->GOLF_COURSE|sport->GOLF", "leisure->PARK&name->*", "natural->BEACH",
+            "tourism->ZOO");
+
+    private static final double MINIMUM_PROPORTION_DEFAULT = 0.01;
+
+    private final double minimumIntersect;
+
+    // List of TaggableFilters where each filter represents all tags for AOIs that should not
+    // overlap
+    private final List<TaggableFilter> aoiFilters = new ArrayList<>();
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public OverlappingAOIPolygonCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.minimumIntersect = (Double) this.configurationValue(configuration,
+                "intersect.minimum.limit", MINIMUM_PROPORTION_DEFAULT);
+        final List<String> aoiFiltersString = (List<String>) configurationValue(configuration,
+                "aoi.tags.filters", AOI_FILTERS_DEFAULT);
+        aoiFiltersString.forEach(string -> this.aoiFilters.add(new TaggableFilter(string)));
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Checks for areas, that are not flagged, and pass any of the TaggableFilters in
+        // aoiFilters. aoiFiltersTest() and aoiFilters is used in place of a single TaggableFilter
+        // so that each filter may be tested separately later.
+        return object instanceof Area && !this.isFlagged(object.getIdentifier())
+                && aoiFiltersTest(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+
+        final Area aoi = (Area) object;
+        final Polygon aoiPolygon = aoi.asPolygon();
+        boolean hasOverlap = false;
+
+        // Set of overlapping area AOIs
+        final Set<Area> overlappingArea = Iterables.stream(object.getAtlas().areasIntersecting(
+                aoiPolygon.bounds(),
+                area -> area.getIdentifier() != aoi.getIdentifier() && area.intersects(aoiPolygon)
+                        && aoiFiltersTest(area) && !this.isFlagged(area.getIdentifier())))
```

Comment:
It might be worthwhile moving the `isFlagged` check before the intersects check.